go/ql/lib/semmle/go/frameworks/Fasthttp.qll

```diff
@@ -0,0 +1,456 @@
+      or
+      methodName = ["SetContentType", "SetContentTypeBytes", "Success", "SuccessString"] and
+      result = "content-type"
+    }
+
+    override DataFlow::Node getValue() {
+      if methodName = ["SetContentType", "SetContentTypeBytes", "Success", "SuccessString"]
+      then result = this.getArgument(0)
+      else result = this.getArgument(1)
+    }
+
+    override Http::ResponseWriter getResponseWriter() {
+      result.(ResponseWriter).getAHeaderObject() = this
+    }
+  }
+
+  private class ResponseWriter extends Http::ResponseWriter::Range {
+    SsaWithFields v;
+
+    ResponseWriter() {
+      this = v.getBaseVariable().getSourceVariable() and
+      (
+        v.getType().hasQualifiedName(packagePath(), ["Response", "ResponseHeader"]) or
+        v.getType().(PointerType).getBaseType().hasQualifiedName(packagePath(), "RequestCtx")
+      )
+    }
+
+    override DataFlow::Node getANode() { result = v.similar().getAUse().getASuccessor*() }
+
+    /** Gets a header object that corresponds to this HTTP response. */
+    DataFlow::MethodCallNode getAHeaderObject() {
+      result.getTarget().getName() =
+        [
+          "Add", "AddBytesK", "AddBytesKV", "AddBytesV", "Set", "SetBytesK", "SetBytesKV",
+          "SetCanonical", "SetContentType", "SetContentTypeBytes", "Success", "SuccessString"
+        ] and
+      this.getANode() = result.getReceiver()
+    }
+  }
+
+  private predicate responseBodyWriterResult(DataFlow::Node src) {
+    exists(Method responseBodyWriter |
+      responseBodyWriter.hasQualifiedName(packagePath(), "Response", "BodyWriter") and
+      src = responseBodyWriter.getACall().getResult(0)
+    )
+  }
+
+  private module ResponseBodyWriterFlow = DataFlow::SimpleGlobal<responseBodyWriterResult/1>;
+
+  private class ResponseBody extends Http::ResponseBody::Range {
+    DataFlow::MethodCallNode call;
+    string methodName;
+
+    ResponseBody() {
+      exists(Method m |
+        m.hasQualifiedName(packagePath(), "Response", methodName) and
+        call = m.getACall() and
+        this = call.getArgument(0)
+        or
+        m.hasQualifiedName(packagePath(), "RequestCtx", ["Success", "SuccessString"]) and
+        this = m.getACall().getArgument(1)
+      ) and
+      methodName =
+        [
+          "AppendBody", "AppendBodyString", "SetBody", "SetBodyRaw", "SetBodyStream",
+          "SetBodyString", "Success", "SuccessString"
+        ]
+      or
+      exists(Method write, DataFlow::CallNode writeCall |
+        write.hasQualifiedName("io", "Writer", "Write") and
+        writeCall = write.getACall() and
+        ResponseBodyWriterFlow::flowsTo(writeCall.getReceiver()) and
+        this = writeCall.getArgument(0)
+      ) and
+      methodName = "BodyWriter"
+    }
+
+    override Http::ResponseWriter getResponseWriter() { result.getANode() = call.getReceiver() }
+
+    override string getAContentType() { result = super.getAContentType() }
```

Comment:
I believe this is saying that you can just delete this predicate, since it just delegates to the `super` one, and if it doesn't exist then the `super` one will be used.